packages/asc-ui/src/components/TextArea/TextArea.tsx

```diff
@@ -0,0 +1,58 @@
+import React, { useImperativeHandle } from 'react'
+import useActionOnEscape from '../../utils/hooks/useActionOnEscape'
+import useFocusOnRender from '../../utils/hooks/useFocusOnRender'
+import TextAreaStyle from './TextAreaStyle'
+
+export interface TextAreaMethods
+  extends React.TextareaHTMLAttributes<HTMLTextAreaElement> {
+  setTextAreaRef?: (ref: React.RefObject<HTMLTextAreaElement>) => void
+}
+
+export interface TextAreaProps extends TextAreaMethods {
+  keepFocus?: boolean
+  blurOnEscape?: boolean
+  focusOnRender?: boolean
+  error?: string
+}
+
+const TextArea = React.forwardRef<HTMLTextAreaElement, TextAreaProps>(
+  ({ blurOnEscape, focusOnRender, value, error, ...props }, externalRef) => {
+    const { onKeyDown } = props
```

Comment:
Can we add this de-structuring to the params of the the function? Like `value`, `error` etc.

Reply:
Done